sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -205,6 +206,20 @@ abstract class QueryTest extends PlanTest {
         planWithCaching)
   }
 
+  /**
+   * Asserts that a given [[Dataset]] will be executed using the given named cache.
+   */
+  def assertCached(query: Dataset[_], cachedName: String, storageLevel: StorageLevel): Unit = {
+    val planWithCaching = query.queryExecution.withCachedData
+    val matched = planWithCaching.collectFirst { case cached: InMemoryRelation =>
+      val cacheBuilder = cached.asInstanceOf[InMemoryRelation].cacheBuilder
+      cacheBuilder.tableName.get == cachedName && cacheBuilder.storageLevel == storageLevel
+    }.nonEmpty
+
+    assert(matched, s"Expected query plan to hit cache $cachedName with storage " +
+      s"Level $storageLevel, but it doesn't.")
```

Comment:
`s"Level` -> `s"level`? Currently, this says `with storage Level ...`.

Reply:
arr. yes. i missed this. I just changed it to lowercase.